### PR TITLE
documentation: Document DOOD

### PR DESCRIPTION
**Pull request by housel**

### CODE REVIEW

File: documentation/library-reference/source/dood/index.rst [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+Tables
+------
+
+.. class:: <dood-lazy-symbol-table>
+
+   :superclasses: :class:`<dood-lazy-key-table>`
+
+   Provide a mechanism for indexes. The keys are symbols and are
+   loaded lazily using a binary search. This is known to be an
+   inferior layout strategy and will be replaced by b*-tree's in the
```
Comment (cgay) [inline documentation/library-reference/source/dood/index.rst:303]: Probably need a backslash before the asterisk.

File: documentation/library-reference/source/dood/index.rst [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+       required-init-keyword: previous:;
+     weak slot computation-type :: false-or(<type-estimate>) = #f,
+       reinit-expression: #f;
+   end dood-class;
+
+Reading
+-------
+
+Internally DOOD loads objects by instantiation and slot assignment. An
+object is instantiated via allocate, which returns an uninitialized
```
Comment (cgay) [inline documentation/library-reference/source/dood/index.rst:258]: I think `allocate` is not defined anywhere.

**Review by cgay — APPROVED** [review]
Very nice.

File: documentation/library-reference/source/dood/index.rst [hunk 3]
```diff
@@ -0,0 +1,527 @@
...
+     slot rectangle-width :: <integer>,
+       required-init-keyword: width:;
+     weak rectangle-area :: <integer>;
+   end dood-class;
+
+   define method dood-reinitialize (dood :: <dood>, object :: <rectangle>)
+     next-method();
+     rectangle-area(object)
+       := rectange-height(object) * rectange-width(object);
+   end method;
```
Comment (pedro-w) [inline documentation/library-reference/source/dood/index.rst:292]: Typo: rectange ?

**Pull request merged**